pkg/remote/kyma_synchronization_context.go

```diff
@@ -87,7 +87,9 @@ func RemoveFinalizerFromRemoteKyma(
 		return err
 	}
 
-	controllerutil.RemoveFinalizer(remoteKyma, shared.KymaFinalizer)
+	for _, finalizer := range remoteKyma.Finalizers {
+		controllerutil.RemoveFinalizer(remoteKyma, finalizer)
+	}
```

Comment:
Just for clarification: Is this change needed because of your other changes? Since you did not introduce any new finalizers. This is kinda like a purge mechanism, if I got it correct.

Reply:
This is not related to my change, but we don't sync the finalizers to the remote kyma. I added this just to remove any finalizers on the remote kyma if there was any added by the user